src/DataSourceProvider.vue

```diff
@@ -1,19 +1,368 @@
 <template>
-  <div>
-    <h1>Your fliplet-widget-data-source-provider widget will be added here.</h1>
-    <First-Component></First-Component>
-  </div>
+  <section>
+    <div class="data-source-title">
+      <strong>{{ widgetData.dataSourceTitle || 'Select a data source' }}</strong>
+    </div>
+
+    <div v-if="isLoading" class="spinner-holder animated">
+      <div class="spinner-overlay">Loading...</div>
+    </div>
+
+    <div v-else-if="hasError" class="alert alert-danger" role="alert">
+      {{ errorMessage }}
+    </div>
+
+    <div v-else class="main-data-source-provider">
+      <section class="data-source-selector">
+
+        <div v-if="dataSources.length && changeDataSource">
+
+          <Select2
+            :dataSources="dataSources"
+            :selectedDataSource="selectedDataSource"
+            :customOptionView="formatDataSourceOption"
+            :customSearch="customDataSourceSearch"
+            :optionLabelKey="'name'"
+            :optionValueKey="'id'"
+            :selectWithGroups="!!allDataSources.length"
+            @selectDataSource="setDataSource">
+          </Select2>
+
+          <span @click="onDataSourceCreate" class="btn-link create-data-source">Create new data source</span>
+
+          <div class="checkbox checkbox-icon">
+            <input @change="showAllDataSources" :checked="showAll" type="checkbox" name="showAll" id="showAll" />
```

Comment:
Use a watcher instead.

```suggestion
            <input :checked="showAll" type="checkbox" name="showAll" id="showAll" />
```